### PR TITLE
Switch tests to shift-parser-expectations

### PR DESCRIPTION
**Pull request by bakkot**
This PR:

- Modifies the existing test262-parser-tests test to additionally check ASTs and metadata against those in the just-published shift-parser-expectations package
- Modifies the test harness to assert tests in this project are not duplicated upstream in shift-parser-expectations
- Adds a script to automatically remove tests in this project which are so duplicated
- Removes those tests (using that script and `eslint --fix`, so this adds a bit of whitespace)
- Modifies the eslintrc for tests to allow unused variables, since many of the files are now effectively empty but we don't want to be constantly adding and removing helpers.

### CODE REVIEW

**Comment by bakkot** [comment]
Also now removes some apparently dead code in early-error-state.js. I don't know what that's about - coverage complains it's never hit.

File: test/expressions/grouped-expressions.js [hunk 1]
```diff
@@ -22,52 +22,13 @@ let testParseFailure = require('../assertions').testParseFailure;
...
-      right: { type: 'IdentifierExpression', name: 'a' },
-    });
 
-    testParse('(a, 0)', expr, {
-      type: 'BinaryExpression',
-      left: { type: 'IdentifierExpression', name: 'a' },
-      operator: ',',
-      right: { type: 'LiteralNumericExpression', value: 0 },
-    });
+    // grouped expression than cannot be binding element or assignment target
```
Comment (michaelficarra) [inline test/expressions/grouped-expressions.js]: Can you go through and remove/fix lonely comments like this one?

File: test/assertions.js [hunk 2]
```diff
@@ -7,6 +10,17 @@ let SHIFT_SPEC = require('shift-spec').default;
 let Parser = require('../dist/parser').GenericParser;
 let EarlyErrorChecker = require('../dist/early-errors').EarlyErrorChecker;
 
+let expectationsDir = 'node_modules/shift-parser-expectations/expectations';
```
Comment (michaelficarra) [inline test/assertions.js:13]:
```js
let expectationsDir = require.resolve('shift-parser-expectations') + '/expectations';
```

It should ideally use `path.join`, but meh.

File: test/assertions.js [hunk 3]
```diff
@@ -7,6 +10,17 @@ let SHIFT_SPEC = require('shift-spec').default;
...
+
+
+function checkPassNotExists(program, isModule) {
+  let digest = crypto.createHash('sha256').update(normalize(program, isModule)).digest('hex');
```
Comment (michaelficarra) [inline test/assertions.js]: Extract this and the substring to its own function.

Comment (bakkot) [reply test/assertions.js]: Done, though it seems slightly silly.

File: test/expressions/grouped-expressions.js (hunk 1 above)
Comment (bakkot) [reply test/expressions/grouped-expressions.js]: Done.

File: scripts/remove-upstreamed-tests.js [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+const crypto = require('crypto');
+const reducer = require('shift-reducer');
+const normalize = require('normalize-parser-test').default;
+const { parseScriptWithLocation, parseModuleWithLocation } = require('shift-parser');
+
+const expectationsDir = 'node_modules/shift-parser-expectations/expectations';
+
+function testExistsUpstream(src, isModule) {
+  const digest = crypto
+    .createHash('sha256')
```
Comment (michaelficarra) [inline scripts/remove-upstreamed-tests.js]: Here, too.

**Review by michaelficarra — APPROVED** [review]

**Pull request merged**